ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx

```diff
@@ -450,13 +459,26 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
   };
 
   useEffect(() => {
-    if (!socketConnected && authActive) {
-      startCommunicationDetectionTimeout(socketConnectionTimeoutRef, setSocketConnectionSnackbarOpen);
+    if (nnou(uiConfig.xml)) {
+      const storedEditMode = getStoredEditModeChoice(user.username, uuid);
+      const storedHighlightMode = getStoredHighlightModeChoice(user.username, uuid);
+      const storedPaddingMode = getStoredEditModePadding(user.username);
+      dispatch(initPreviewConfig({ configXml: uiConfig.xml, storedEditMode, storedHighlightMode, storedPaddingMode }));
+    }
+  }, [uiConfig.xml, user.username, uuid, dispatch]);
+
+  useEffect(() => {
+    if (!socketConnected && authActive && xbDetectionTimeoutMs) {
+      startCommunicationDetectionTimeout(
```

Comment:
The socket connection should stay as it was. Only XB detection should take this change.